lib/spack/spack/installer.py

```diff
@@ -400,7 +400,11 @@ def dump_packages(spec, path):
 
             # There's no provenance installed for the source package.  Skip it.
             # User can always get something current from the builtin repo.
-            if not os.path.isdir(source_repo_root):
+            # If it's external, skip it because it either is using the builtin
+            # repo or the provenance is already in place.
+            # The provenance could exist if the external was installed by a
+            # separate Spack instance not configured as an upstream.
```

Comment:
Based on https://github.com/spack/spack/pull/16954#discussion_r435592514, I think that even if the appropriate package.py files are placed in the repository, we don't currently set the namespace right.

Also, based on that comment, I think that once we get the namespace right, it won't matter whether the other Spack instance is marked as an upstream or not.

So in short, I don't think this comment is correct. Do my above arguments make sense, or am I missing something?